### PR TITLE
INTYGFV-16168: Implementation of getHealthCareUnit.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

File: integration/hsa-integration-intyg-proxy-service/src/test/java/se/inera/intyg/infra/integration/intygproxyservice/services/organization/HsaIntegrationOrganizationServiceTest.java [hunk 1]
```diff
@@ -38,9 +40,26 @@ class HsaIntegrationOrganizationServiceTest {
     private static final String UNIT_HSA_ID = "unitHsaId";
     @Mock
     private GetUnitService getUnitService;
+    @Mock
+    private GetHealthCareUnitService getHealthCareUnitService;
     @InjectMocks
     private HsaIntegrationOrganizationService hsaIntegrationOrganizationService;
 
+    @Nested
```
Comment (mikaelaromild1) [inline integration/hsa-integration-intyg-proxy-service/src/test/java/se/inera/intyg/infra/integration/intygproxyservice/services/organization/HsaIntegrationOrganizationServiceTest.java:48]: Should we add tests for when null is returned?

**Review by mikaelaromild1 — CHANGES_REQUESTED** [review]
Looks good, just a small suggestion :)

**Review by mikaelaromild1 — APPROVED** [review]
Nice!
